src/_nebari/template/stages/07-kubernetes-services/modules/kubernetes/services/jupyterhub/files/jupyterhub/03-profiles.py

```diff
@@ -311,6 +311,61 @@ def configure_user(username, groups, uid=1000, gid=100):
     }
 
 
+def profile_argo_token(groups):
+    # TODO: create a more robust check user's Argo-Workflow role
+
+    domain = z2jh.get_config("custom.external-url")
+
+    ADMIN = "admin"
+    DEVELOPER = "developer"
+    ANALYST = "analyst"
+
+    base = "argo-"
+
+    if ANALYST in groups:
+        argo_sa = base + "view"
+    if DEVELOPER in groups:
+        argo_sa = base + "dev"
+    if ADMIN in groups:
+        argo_sa = base + "admin"
+    else:
+        return {}
```

Comment:
This seems a bit clunky.  Not sure if there's a better way.

Reply:
I agree! The difficult part is that `roles` aren't available in the current `spawner.user.auth_state`. I think the other alternative is to perform a keycloak API call. We would probably also need to add keycloak secrets to the `hub` deployment. 

I have this particular enhancement in [this issue](https://github.com/nebari-dev/nebari/issues/1838).